### PR TITLE
Improved error logging and feedback by use of Sentry

### PR DESCRIPTION
**Pull request by mjvkeulen**
Added [sentry](http://sentry.io) to compliment our logging tooling. 

- Backend error logging (compatible with celery).
- Frontend error logging.
- Updated frontend dependencies.
- Coupled user feedback to frontend state and trace.

Partially solves #277, but not our generic usage logging.  

image
image

### CODE REVIEW

File: src/vue/src/api/connection.js [hunk 1]
```diff
@@ -1,28 +1,56 @@
...
 
-store.dispatch('connection/setupConnectionInterceptors', { connection: conn })
-store.dispatch('connection/setupConnectionInterceptors', { connection: connRefresh, isRefresh: true })
-store.dispatch('connection/setupConnectionInterceptors', { connection: connFile })
-store.dispatch('connection/setupConnectionInterceptors', { connection: connFileEmail })
+const connSentry = axios.create({
+    /* /api/0/projects/{organization_slug}/{project_slug}/ */
+    baseURL: `https://sentry.io/api/0/projects/${CustomEnv.SENTRY_ORGANIZATION_SLUG}/${CustomEnv.SENTRY_PROJECT_SLUG}/`,
+    headers: {
+        Authorization: `DSN ${CustomEnv.SENTRY_DSN}`,
```
Comment (Larspolo) [inline src/vue/src/api/connection.js]: Doesnt this leak the key?

Comment (mjvkeulen) [reply src/vue/src/api/connection.js]: Its public

File: src/vue/src/views/ErrorPage.vue [hunk 2]
```diff
@@ -32,9 +48,7 @@ export default {
             default: 'Unknown Error',
         },
         description: {
-            default: 'We are sorry, but an unkown error has brought you here. '
-                + 'Please use the feedback button at the bottom of the page '
-                + 'to get in touch with us.',
+            default: 'We are sorry, but an unkown error has brought you here.',
```
[the remark this answers is not in the document]
Comment (mjvkeulen) [reply src/vue/src/views/ErrorPage.vue]: Why not keep the apology?

[the remark this answers is not in the document]
Comment (Larspolo) [reply src/vue/src/views/ErrorPage.vue]: It appears when reloading the page / going to error manually. Its a bit weird imo. Also the n in unknown

File: src/vue/src/components/sentry/SentryFeedbackForm.vue [hunk 3]
```diff
@@ -0,0 +1,112 @@
+<template>
```
Comment (engelhamer) [inline src/vue/src/components/sentry/SentryFeedbackForm.vue]: I'll take a look at the styling of this form at a later moment.

File: src/vue/src/helpers/sentry.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import * as Sentry from '@sentry/browser'
+import * as Integrations from '@sentry/integrations'
+// import router from '@/router'
```
Comment (engelhamer) [inline src/vue/src/helpers/sentry.js]: Lots of disabled code here.

File: src/vue/src/views/ErrorPage.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (engelhamer) [reply src/vue/src/views/ErrorPage.vue]: When a user manually navigates to this page I agree that they should not be presented with this message. Ideally they would be redirected to the base url (which would mean home page or landing page).

File: src/vue/src/views/Home.vue [hunk 5]
```diff
@@ -94,6 +107,15 @@ export default {
             .then((deadlines) => { this.deadlines = deadlines })
     },
     methods: {
+        // TODO SENTRY remove both errors and button
```
Comment (engelhamer) [inline src/vue/src/views/Home.vue]: Be careful that no colons follow `TODO` here.

File: src/django/VLE/tasks/grading.py [hunk 6]
```diff
@@ -56,3 +56,8 @@ def send_journal_grade_to_LMS(journal_pk):
         return "Journal {} grade was succesfully sent to the LMS".format(journal_pk)
 
     return "Journal {} grade could not be sent to the LMS. Response was: {}".format(journal_pk, response)
+
+
+@shared_task
+def trigger_error_task():
```
Comment (engelhamer) [inline src/django/VLE/tasks/grading.py]: I reckon this should also be removed? Don't see a `TODO Sentry` here.

File: src/vue/src/helpers/sentry.js (hunk 4 above)
Comment (mjvkeulen) [reply src/vue/src/helpers/sentry.js]: I was hoping to inspire some discussion on any error filtering before we log them

File: src/vue/src/views/ErrorPage.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (mjvkeulen) [reply src/vue/src/views/ErrorPage.vue]: We can redirect away from the error page on refresh, but anyone who manually (by editing the url) navigates here can click on `Home` imo.

**Comment by engelhamer** [comment]
Made the error page consistent with the overall style:

**Pull request merged**